
# Python Set Comprehensions

Python set comprehensions are similar to list comprehensions and allow a programmer to quickly create sets using simple, concise, and readable code.

To demonstrate Set Comprhensions we are going to manipulate and work with Countries. To do this we will use the pycountry library.


## Step #1

### Install the pycountry library

In [18]:
pip install pycountry

Note: you may need to restart the kernel to use updated packages.


## Step #2

#### Next we will import pycountry into the project and print a count of how many countries are in the list



In [16]:
import pycountry

len(pycountry.countries)

249

## Step #3

### Next we will create a list of all 249 countries and print the first one to show its format

In [50]:
countries = list(pycountry.countries)
countries[1]

Country(alpha_2='AF', alpha_3='AFG', name='Afghanistan', numeric='004', official_name='Islamic Republic of Afghanistan')



## Step #4

### Next we create a list containing every word used in any of the official_name attributes from the list of countries

In [94]:
official_names = []
for x in countries:        
    if hasattr(x, 'official_name'):
        official_names.extend(x._fields['official_name'].split())
official_names


['Islamic',
 'Republic',
 'of',
 'Afghanistan',
 'Republic',
 'of',
 'Angola',
 'Republic',
 'of',
 'Albania',
 'Principality',
 'of',
 'Andorra',
 'Argentine',
 'Republic',
 'Republic',
 'of',
 'Armenia',
 'Republic',
 'of',
 'Austria',
 'Republic',
 'of',
 'Azerbaijan',
 'Republic',
 'of',
 'Burundi',
 'Kingdom',
 'of',
 'Belgium',
 'Republic',
 'of',
 'Benin',
 'Bonaire,',
 'Sint',
 'Eustatius',
 'and',
 'Saba',
 "People's",
 'Republic',
 'of',
 'Bangladesh',
 'Republic',
 'of',
 'Bulgaria',
 'Kingdom',
 'of',
 'Bahrain',
 'Commonwealth',
 'of',
 'the',
 'Bahamas',
 'Republic',
 'of',
 'Bosnia',
 'and',
 'Herzegovina',
 'Republic',
 'of',
 'Belarus',
 'Plurinational',
 'State',
 'of',
 'Bolivia',
 'Federative',
 'Republic',
 'of',
 'Brazil',
 'Kingdom',
 'of',
 'Bhutan',
 'Republic',
 'of',
 'Botswana',
 'Swiss',
 'Confederation',
 'Republic',
 'of',
 'Chile',
 "People's",
 'Republic',
 'of',
 'China',
 'Republic',
 'of',
 'Côte',
 "d'Ivoire",
 'Republic',
 'of',
 'Cameroon',
 'Repu



## Step #4

### Last, we will create a set containing all unique words used to form a countries official name.

In [95]:
a = {x for x in official_names}
a

{'(Dutch',
 'Administrative',
 'Afghanistan',
 'Africa',
 'Albania',
 'Algeria',
 'America',
 'Andorra',
 'Angola',
 'Arab',
 'Arabia',
 'Argentine',
 'Armenia',
 'Austria',
 'Azerbaijan',
 'Bahamas',
 'Bahrain',
 'Bangladesh',
 'Belarus',
 'Belgium',
 'Benin',
 'Bhutan',
 'Bolivarian',
 'Bolivia',
 'Bonaire,',
 'Bosnia',
 'Botswana',
 'Brazil',
 'Britain',
 'British',
 'Bulgaria',
 'Burundi',
 'Cabo',
 'Cambodia',
 'Cameroon',
 'Chad',
 'Chile',
 'China',
 'Colombia',
 'Commonwealth',
 'Comoros',
 'Confederation',
 'Congo',
 'Costa',
 'Croatia',
 'Cuba',
 'Curaçao',
 'Cyprus',
 'Czech',
 'Côte',
 'Democratic',
 'Denmark',
 'Djibouti',
 'Dominica',
 'Duchy',
 'Eastern',
 'Ecuador',
 'Egypt',
 'El',
 'Equatorial',
 'Eritrea',
 'Estonia',
 'Eswatini',
 'Ethiopia',
 'Eustatius',
 'Federal',
 'Federated',
 'Federative',
 'Fiji',
 'Finland',
 'French',
 'Gabonese',
 'Gambia',
 'Germany',
 'Ghana',
 'Grand',
 'Great',
 'Guatemala',
 'Guinea',
 'Guinea-Bissau',
 'Guyana',
 'Haiti',
 'Hashemit